### PR TITLE
[SPARK-42621][PS] Add inclusive parameter for pd.date_range

### PR DESCRIPTION
**Pull request by dzhigimont**
### What changes were proposed in this pull request?
Add inclusive parameter for pd.date_range to support the pandas 2.0.0

### Why are the changes needed?
When pandas 2.0.0 is released, we should match the behavior in pandas API on Spark.

### Does this PR introduce any user-facing change?
yes, the API changes

Before:
ps.date_range(start='2017-01-01', end='2017-01-04', closed=None)

After:
ps.date_range(start='2017-01-01', end='2017-01-04', inclusive="both")

### How was this patch tested?
Unit tests were updated

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
cc @itholic @zhengruifeng @xinrong-meng @Yikun if you find some time to review.

File: python/pyspark/pandas/namespace.py [hunk 1]
```diff
@@ -1782,12 +1780,8 @@ def date_range(
         Normalize start/end dates to midnight before generating date range.
     name : str, default None
         Name of the resulting DatetimeIndex.
-    closed : {None, 'left', 'right'}, optional
-        Make the interval closed with respect to the given frequency to
-        the 'left', 'right', or both sides (None, the default).
-
-        .. deprecated:: 3.4.0
-
+    inclusive : {"both", "neither", "left", "right"}, default "both"
```
Comment (xinrong-meng) [inline python/pyspark/pandas/namespace.py:1783]: nit: Shall we add `versionadded`?

Comment (dzhigimont) [reply python/pyspark/pandas/namespace.py:1785]: Sure, versionadded has been added

File: python/pyspark/pandas/tests/test_namespace.py [hunk 2]
```diff
@@ -221,13 +221,13 @@ def test_date_range(self):
...
-            pd.date_range(start="2017-01-01", end="2017-01-04", closed="left"),
+            ps.date_range(start="2017-01-01", end="2017-01-04", inclusive="left"),
+            pd.date_range(start="2017-01-01", end="2017-01-04", inclusive="left"),
         )
 
         self.assert_eq(
-            ps.date_range(start="2017-01-01", end="2017-01-04", closed="right"),
-            pd.date_range(start="2017-01-01", end="2017-01-04", closed="right"),
+            ps.date_range(start="2017-01-01", end="2017-01-04", inclusive="right"),
+            pd.date_range(start="2017-01-01", end="2017-01-04", inclusive="right"),
```
Comment (itholic) [inline python/pyspark/pandas/tests/test_namespace.py:230]: Can we also have a tests for "both", "neither" and negative case.

Comment (dzhigimont) [reply python/pyspark/pandas/tests/test_namespace.py:230]: Done

**Comment by dzhigimont** [comment]
@itholic Could you please reopen the PR and remove the stale label?

**Comment by HyukjinKwon** [comment]
@itholic please review this

File: python/pyspark/pandas/namespace.py [hunk 3]
```diff
@@ -1875,36 +1872,31 @@ def date_range(
                    '2019-01-31'],
                   dtype='datetime64[ns]', freq=None)
 
-    `closed` controls whether to include `start` and `end` that are on the
+    `inclusive` controls whether to include `start` and `end` that are on the
     boundary. The default includes boundary points on either end.
```
Comment (itholic) [inline python/pyspark/pandas/namespace.py:1876]: nit: to match the pandas docs

```suggestion
    `inclusive` controls whether to include `start` and `end` that are on the
    boundary. The default, “both”, includes boundary points on either end.
```

**Review by itholic — COMMENTED** [review]
Looks pretty good otherwise.

**Review by itholic — APPROVED** [review]

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Comment by itholic** [comment]
Thanks, @dzhigimont for working on this
